main.go

```diff
@@ -36,10 +37,22 @@ func run() int {
 		}
 
 		// Print error if not due to help request.
-		logger.Error("Failed to load configuration", "error", err)
+		fmt.Fprintf(os.Stderr, "Failed to load configuration: %v", err)
 		return 1
 	}
-	defer cleanupWorkspace(logger, cfg)
+	defer cleanupWorkspace(cfg)
+
+	// Initialize a structured logger that writes to both stdout and the
+	// rotating log file.
+	logFile := &lumberjack.Logger{
+		Filename:   filepath.Join(cfg.LogDir, LogFilename),
+		MaxSize:    100,
+		MaxBackups: 7,
+		MaxAge:     28,
+		Compress:   true,
+	}
+	multiWriter := io.MultiWriter(os.Stdout, logFile)
+	logger := slog.New(slog.NewTextHandler(multiWriter, nil))
```

Comment:
We can still use the logger during cleanup -- we just need to put the `defer` after we create the logger.